inquire/src/prompts/path_select/path_entry.rs

```diff
@@ -0,0 +1,175 @@
+//! Path entries
+use super::{PathSelectionMode, PathSortingMode};
+use crate::{InquireError, };
+use std::{
+    cmp, 
+    convert::TryFrom,
+    ffi::OsStr,
+    fmt, fs,
+    ops::Deref,
+    path::{Path, PathBuf}
+};
+
+/// A path with cached information
+#[derive(Clone, Debug, Hash)]
+pub struct PathEntry {
+    /// The (owned) [path](PathBuf).
+    ///
+    /// Corresponds to the target if this is a symlink.  
+    pub path: PathBuf,
+    /// The [file type](fs::FileType)
+    pub file_type: fs::FileType,
+    /// The original symlink path.
+    pub symlink_path_opt: Option<PathBuf>,
+    /// The file size in bytes.
+    /// 
+    /// Corresponds to the target file size if this is a symlink.
+    pub size: u64,
+    /// If set, this prefix will be stripped from the path during  display.
+    pub strip_prefix_opt: Option<PathBuf>,
+}
+
+impl Eq for PathEntry {}
+
+impl PartialEq for PathEntry {
+    fn eq(&self, other: &Self) -> bool {
+        self.path.eq(&other.path)
+    }
+}
+
+impl AsRef<Path> for PathEntry {
+    fn as_ref(&self) -> &Path {
+        self.path.as_path()
+    }
+}
+
+impl Deref for PathEntry {
+    type Target = fs::FileType;
+    fn deref(&self) -> &Self::Target {
+        &self.file_type
+    }
+}
+
+impl fmt::Display for PathEntry {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let path = if let Some(strip_prefix) = self.strip_prefix_opt.as_ref() {
+            self.path.strip_prefix(strip_prefix)
+                .unwrap_or_else(|_| &self.path)
+                .to_string_lossy()
+        } else {
+            self.path.to_string_lossy()
+        };
+        use humansize::FormatSize;
+        let size_formatting = if cfg!(windows) {
+            humansize::WINDOWS
+        } else {
+            humansize::DECIMAL
+        };
+        let size_string = self.size.format_size(size_formatting); 
+        if let Some(symlink_path) = self.symlink_path_opt.as_ref() {
+            write!(f, "{} -> {path} ({size_string})", symlink_path.to_string_lossy())
+        } else if self.is_dir() {
+            write!(f, "(dir) {path} ({size_string})")
```

Comment:
Wouldn't this always be 0?

Reply:
As I coded it, yes. I wrote it using the `size_string` here for consistency to avoid having to update the 0 value in two places. 

Using of the 0 value for directories was a workaround because for me on Linux, directory sizes always show up as 4096. I am not sure if there are OSes where the `Metadata::len` function returns something more correct, but I wanted to leave it open to the possibility in case you decide to provide an actual accurate size with an OS-specific compiler flag.